.github/workflows/dotnet-codestyle.yml

```
@@ -0,0 +1,34 @@
+name: .NET Codestyle
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [ master ]
+    paths:
+      - '**/*.cs'
+      - '.github/workflows/dotnet-codestyle.yml'
+  pull_request:
+    branches: [ master, 'v[0-9]+.[0-9]+' ]
+    paths:
+      - '**/*.cs'
+      - '.github/workflows/donet-codestyle.yml'
+
+jobs:
+  check-dotnet-codestyle:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
```

`v2` is being deprecated so now's a good time to update

```suggestion
    - uses: actions/checkout@v3
```